app/controllers/PayPal.scala

```diff
@@ -0,0 +1,64 @@
+
+package controllers
+
+import actions.CustomActionBuilders
+import actions.CustomActionBuilders.AuthRequest
+import assets.AssetsResolver
+import com.typesafe.scalalogging.LazyLogging
+import play.api.libs.json.Json.toJson
+import play.api.libs.json._
+import play.api.mvc._
+import services.paypal.{PayPalBillingDetails, PayPalServiceProvider, Token}
+import services.{PayPalService, TestUserService}
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.{ExecutionContext, Future}
+
+class PayPal(
+    actionBuilders: CustomActionBuilders,
+    assets: AssetsResolver,
+    payPalServiceProvider: PayPalServiceProvider,
+    testUsers: TestUserService,
+    components: ControllerComponents
+)(implicit val ec: ExecutionContext) extends AbstractController(components) with LazyLogging {
+
+  import actionBuilders._
+
+  // Json readers & writers.
+  implicit val formatsToken = Json.format[Token]
```

Comment:
Json readers/writers are normally better placed in the companion objects (`Token` and `PaypalBillingDetails`).  Also, the rest of this project uses circe for json conversion - it would be nice to be consistent throughout the project.